### PR TITLE
Update ssl-config-core 0.6.0 (was 0.4.3)

### PR DESCRIPTION
**Pull request by mkurz**
Made #30871 binary compatibile by making `validateDefaultTrustManager` a no-op.
Unblocks https://github.com/playframework/playframework/pull/11081

### CODE REVIEW

**Review by johanandren — APPROVED** [review]
LGTM but would be good with @jrudolph or @raboof s eyes on this one for the dropped parts just to be sure.

**Comment by mkurz** [comment]
Thanks Johan!

**Comment by johanandren** [comment]
Needs some `@nowarn("msg=never used")` since compiler warn fails the build

**Comment by raboof** [comment]
(I think one `@nowarn` is missing an import)

File: project/Dependencies.scala [hunk 1]
```diff
@@ -87,7 +87,7 @@ object Dependencies {
     val reactiveStreams = "org.reactivestreams" % "reactive-streams" % reactiveStreamsVersion // CC0
 
     // ssl-config
-    val sslConfigCore = ("com.typesafe" %% "ssl-config-core" % sslConfigVersion).cross(CrossVersion.for3Use2_13) // ApacheV2
+    val sslConfigCore = "com.typesafe" %% "ssl-config-core" % sslConfigVersion // ApacheV2
```
Comment (raboof) [inline project/Dependencies.scala:90]: neat!

**Review by raboof — APPROVED** [review]
Looks good I think

**Comment by mkurz** [comment]
Force pushed with the import and ran scalafmt, squashed as well. Should be good now.

**Comment by mkurz** [comment]
Green :+1: @johanandren or @raboof can you please merge, I am eager to try the next snapshot, thanks!

**Comment by mkurz** [comment]
:heart:

**Comment by raboof** [comment]
It appears ssl-config 0.4.3 itself is not binary compatible with 0.6.0: in particular, https://github.com/lightbend/ssl-config/pull/302 removes `SSLLooseConfig.allowWeakProtocols`, which is still called by play-ws 2.1.2 and earlier.

Not sure what would be the wisest way to deal with that. Perhaps we could backport dropping the parser-combinator library dependency except without breaking binary compatibility so badly and release that as 0.4.4, or selectively fix some of the more prominent breakages in 0.6.1 and leave the rest. WDYT?

**Comment by mkurz** [comment]
How about releasing play-ws 2.1.3 with upgraded ssl-config 0.6.0 and simultaneously Play 2.8.14 with that play-ws 2.1.3
I actually already merged https://github.com/playframework/play-ws/pull/638 for that.

However if that is not the best way, I would just re-add breaking stuff to a ssl-config 0.6.1 release.

**Comment by mkurz** [comment]
Also see [11109](https://github.com/playframework/playframework/pull/11109)

**Comment by mkurz** [comment]
@raboof How do you want to handle this?

**Comment by mkurz** [comment]
Hey @raboof and @SethTisue, what do you think, how should we handle https://github.com/akka/akka/pull/31046#issuecomment-1021176742?

**Comment by raboof** [comment]
If there are no longer any known supported projects that use the now-removed API's, I think I'm OK with not making further changes. What do you think?

**Comment by SethTisue** [comment]
(Sorry, I don’t think I have useful input here.)

**Comment by mkurz** [comment]
The easiest thing IMHO is to, like suggested, release play-ws 2.1.3 and Play 2.8.14, both with upgraded ssl-config (to match the version used in akka), immediately after akka 2.8.19 got released. Like said, I already prepared https://github.com/playframework/play-ws/pull/638 for that.

**Comment by raboof** [comment]
> The easiest thing IMHO is to, like suggested, release play-ws 2.1.3 and Play 2.8.14, both with upgraded ssl-config (to match the version used in akka), immediately after akka 2.8.19 got released. Like said, I already prepared [playframework/play-ws#638](https://github.com/playframework/play-ws/pull/638) for that.

Ah, I didn't realize that hadn't happened yet - I saw 2.1.3 was already out for a while at https://repo1.maven.org/maven2/com/typesafe/play/play-ws-standalone_2.13/ but looking closer indeed that doesn't seem to have the change yet. Is there anything stopping you from already releasing with that change, before we release akka 2.8.19?

**Comment by raboof** [comment]
Now that I understand there are non-ancient projects still calling some of the methods that disappeared in 0.6.0, perhaps we should go back and re-introduce those (as no-ops and marked deprecated) in a config-core 0.6.1.

**Comment by mkurz** [comment]
@raboof That is probably the safest thing to do. I can do that - when do you plan to release akka 2.8.19?

**Comment by mkurz** [comment]
(BTW: The next 2.1.x play-ws release is 2.1.8... Probably a typo in my original comment. Sorry, my mistake. )

**Comment by raboof** [comment]
> I can do that

That would be great!

> when do you plan to release akka 2.8.19?

I don't think we have it planned yet - though 2.8.18 was a while ago already so perhaps we should ;)

**Comment by mkurz** [comment]
Alright, I'll try to do it over the weekend, looking forward to the next akka release :wink:

**Pull request merged**